ansible/ansible-ai-connect-service
# AAP-33634: ModelPipelines: Health endpoint improvements

Jira Issue: https://issues.redhat.com/browse/AAP-33634

## Description
Add/expose "Health Check" for each pipeline.

The "aliases" of "Completions" and "Chat Bot" have been preserved as `model-server` and `chatbot-service` respectfully.

This means we _should not_ have to update our test suites - nor anybody else who depends on the existing names.

## Testing
Run the service locally and check the _health_ endpoint: http://127.0.0.1:8000/check/status/

Example output:
```
{
  "timestamp": "2024-12-06T09:38:32.625598",
  "version": "image-tags-not-defined",
  "git_commit": "git-commit-hash-not-defined",
  "deployed_region": "unknown",
  "status": "error",
  "dependencies": [
    {
      "name": "secret-manager",
      "status": "ok",
      "time_taken": 466.697
    },
    {
      "name": "authorization",
      "status": "ok",
      "time_taken": 0.032
    },
    {
      "name": "db",
      "status": "ok",
      "time_taken": 39.233
    },
    {
      "name": "chatbot-service",
      "status": {
        "provider": "http",
        "models": "unavailable: An error occurred" ## <!-- [manstis] I don't have a public route to the ChatBot backend
      },
      "time_taken": 41.034
    },
    {
      "name": "model-server",
      "status": {
        "provider": "wca",
        "tokens": "ok",
        "models": "ok"
      },
      "time_taken": 3394.077
    },
    {
      "name": "content-match",
      "status": "disabled",
      "time_taken": 0.002
    },
    {
      "name": "playbook-explanation",
      "status": "disabled",
      "time_taken": 0.002
    },
    {
      "name": "playbook-generation",
      "status": "disabled",
      "time_taken": 0.001
    },
    {
      "name": "role-generation",
      "status": "disabled",
      "time_taken": 0.002
    }
  ]
}
```

### Steps to test
1. Pull down the PR
2. Run the service locally
3. Check the _health_ endpoint http://127.0.0.1:8000/check/status/

### Scenarios tested
As above.

## Production deployment
